docs/source/elasticsearch.asciidoc

```diff
@@ -0,0 +1,138 @@
+// SPDX-FileCopyrightText: 2024 Timeseer.AI
+// SPDX-License-Identifier: Apache-2.0
+
+== Elasticsearch
+
+Sources with `type = "elasticsearch"` configure https://www.elastic.co/guide/en/elasticsearch/reference/current/index.html[Elasticsearch] sources.
+
+
+```toml
+[source.<name>]
+type = "elasticsearch"
+scheme = "http://"
+host = "localhost"
+port = 9200
+index = ""
+list_index = ""
+tag_columns = []
+field_columns = []
+metadata_columns = []
+metadata_field_column = ""
+timestamp_column = ""
+search_query = ""
+metadata_query = ""
+metadata_mapping = ""
+metadata_value_mapping = ""
+query_timeout_seconds = 60
+```
+
+`index` is required.
+It refers to the Elasticsearch index.
+
+`list_index` is optional and defaults to `index` if not provided.
+It can be used to refer to a different Elasticsearch index for listing time series and their metadata.
+
+Currently https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl.html[Query DSL] is supported.
```

Comment:
Based on this, it is not clear to me that when I provide a `search_query`, I do not need to provide a `list_index`.